apps/ejabberd/src/mod_http_notification_requestor.erl

```diff
@@ -79,16 +82,43 @@ code_change(_OldVsn, State, _Extra) ->
 %%% Internal functions
 %%%===================================================================
 
-make_req(Connection, Host, Path, Sender, Receiver, Message, Timeout) ->
+make_req(Host, PoolName, Path, Sender, Receiver, Message) ->
+    Pool = mongoose_http_client:get_pool(PoolName),
     Query = <<"author=", Sender/binary, "&server=", Host/binary, "&receiver=", Receiver/binary, "&message=",
-              Message/binary>>,
-    ?INFO_MSG("Making request '~s' for user ~s@~s...", [Path, Sender, Host]),
-    Header = [{<<"Content-Type">>, <<"application/x-www-form-urlencoded">>}],
-    case fusco:request(Connection, <<Path/binary>>, "POST", Header, Query, 2, Timeout) of
-        {ok, {{Code, _Reason}, _RespHeaders, RespBody, _, _}} ->
-            ?INFO_MSG("Request result: ~s: ~p", [Code, RespBody]),
-            ok;
-        Else ->
-            Else
-    end.
+        Message/binary>>,
+    ?INFO_MSG("Making request '~p' for user ~s@~s...", [Sender, Host]),
+    Headers = [{<<"Content-Type">>, <<"application/x-www-form-urlencoded">>}],
+    T0 = os:timestamp(),
+    {Res, Elapsed} = case mongoose_http_client:post(Pool, Path, Headers, Query) of
+                         {ok, _} ->
+                             {ok, timer:now_diff(os:timestamp(), T0)};
+                         {error, Reason} ->
+                             {{error, Reason}, 0}
+                     end,
+    record_result(Host, Res, Elapsed),
+    ok.
 
+ensure_metrics(Host) ->
+    mongoose_metrics:ensure_metric([Host, mod_http_notifications, sent], spiral),
+    mongoose_metrics:ensure_metric([Host, mod_http_notifications, failed], spiral),
+    mongoose_metrics:ensure_metric([Host, mod_http_notifications, response_time], histogram),
+    ok.
+record_result(Host, ok, Elapsed) ->
+    mongoose_metrics:update([Host, mod_http_notifications, sent], 1),
+    mongoose_metrics:update([Host, mod_http_notifications, response_time], Elapsed),
+    ok;
+record_result(Host, {error, Reason}, _) ->
+    mongoose_metrics:update([Host, mod_http_notifications, failed], 1),
+    ?WARNING_MSG("Sending http notification failed: ~p", [Reason]),
+    ok.
+
+%% @doc Convert to binary, strip initial slash (it is added by mongoose_http_client)
+fix_path(P) when is_list(P) ->
+    fix_path(list_to_binary(P));
+fix_path(P) when is_binary(P) ->
+    case P of
+        <<47/integer, R/binary>> ->
```

Comment:
We all know `47` is `/` in ASCII but following syntax is probably more readable:
```erlang
<<$/, R/binary>>
```
or
```erlang
<<"/", R/binary>>
```